homeassistant/components/switchbot/coordinator.py

```diff
@@ -0,0 +1,59 @@
+"""Provides the switchbot DataUpdateCoordinator."""
+from __future__ import annotations
+
+from asyncio import Lock
+from datetime import timedelta
+import logging
+
+import switchbot  # pylint: disable=import-error
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SwitchbotDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching switchbot data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        *,
+        update_interval: int,
+        api: switchbot,
+        retry_count: int,
+        scan_timeout: int,
+        api_lock: Lock,
+    ) -> None:
+        """Initialize global switchbot data updater."""
+        self.switchbot_api = api
+        self.retry_count = retry_count
+        self.scan_timeout = scan_timeout
+        self.update_interval = timedelta(seconds=update_interval)
+
+        super().__init__(
+            hass, _LOGGER, name=DOMAIN, update_interval=self.update_interval
+        )
+
+        self.api_lock = api_lock
```

Comment:
Store this in `hass.data[DOMAIN][BTLE_LOCK]` since it likely needs to share the same lock as the config flow.

Reply:
Done.